src/components/shared/GoogleAutoComplete/GoogleAutoComplete.tsx

```diff
@@ -25,36 +21,30 @@ class GoogleAutoComplete extends React.Component<Props, any> {
   autocomplete: AutocompleteInterface;
   
   componentDidMount() {
-    if (!this.props.inputRef.current) return;
+    const input = document.getElementById('locationQuery') as HTMLInputElement;
```

Comment:
On using [`document.getElementById` in React](https://www.javascriptstuff.com/use-refs-not-ids/):

> This would actually work. **At first.**

Aside from reusability, using `getElementById` makes regular use more complicated; you need to know to include a child `input` with a specific ID, and to only include one on a page.

It looks like the issue here was providing styling as desired for the `input` itself, is that correct? If so, I think using `children` is a good approach, and I'd just want to change the way autocomplete binds to the input.

Since `google.maps.places.Autocomplete` needs the reference to the `input`, we do need to pass that in. Think usage could come out looking something like:

    <GoogleAutoComplete>
      { bindInput => <input ref={bindInput} /> }
    </GoogleAutoComplete>

`bindInput` here would mostly look like `componentDidMount`

Reply:
i didn't imagine a situation where we would have more than one input with the same id on the same page, but I see what you're getting at.

At first the biggest issue was the styling, I didn't want to create a new wrapper around `GoogleAutoComplete` if we didn't have to.  Another reason for the refactor was that the ref being created and passed around was difficult to keep track of. 

Im not completely sure what you mean in your example at the bottom